main.go

```diff
@@ -44,13 +47,25 @@ func main() {
 	flag.Parse()
 	beforeRef := "master"
 	afterRef := "HEAD"
+	if flagCl != nil {
```

Comment:
Do `if flag.NArg() > 0 { log.Fatal("-cl NNN is incompatible with ref arguments") }` here. Then you can remove the flagCL check below. To actually correctly compare a CL against a commit that isn't one of its parents, you need to rebase/cherry-pick, and that is definitely beyond the scope of this script.

Reply:
Fixed